### PR TITLE
Task steps for Snapshots and Clones using Kubectl

### PR DESCRIPTION
**Pull request by ChandanSagar**
Signed-off-by: Chandan Sagar Pradhan chandan.pradhan@mayadat.io

### CODE REVIEW

File: docs/tasks_snapclones.md [hunk 1]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
 ------
 
 ## Snapshots using Kubectl
+For taking a snapshot, you must have a PersistentVolumeClaim for which you need to take snapshot. Once you delpoy the application , it's
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: delpoy = deploy
it"s = PersistentVolumeClaim's

File: docs/tasks_snapclones.md [hunk 2]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
 ------
 
 ## Snapshots using Kubectl
+For taking a snapshot, you must have a PersistentVolumeClaim for which you need to take snapshot. Once you delpoy the application , it's 
+PVC will be created by the name specified in the application deployment yaml.
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: The PVC is created with the name ...

File: docs/tasks_snapclones.md [hunk 3]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+Once the application is running, you take a snapshot. To take a snapshot you must reference the PersistentVolumeClaim name in the snapshot specification that references the data for which you want to take snapshot. An example of `snapshot.yaml` file given below.
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: ... ,you can take a snapshot.
you must reference = you must refer
that references = that refers
file is given below.

File: docs/tasks_snapclones.md [hunk 4]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+  namespace: default
+spec:
+  persistentVolumeClaimName: demo-vol1-claim
+```
+Here snapshot-demo is the snapshot name and demo-vol1-claim is the PVC name for which you are taking the snapshot.
+You can now take a snapshot using the command below.
+```
+kubectl apply -f snapshot.yaml
+```
+The command below will show the snapshots that has already been taken.
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: Using the command below will display the snapshots that are already taken.

File: docs/tasks_snapclones.md [hunk 5]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+```
+kubectl get volumesnapshot
+```
+For more details on snapshots please click [here](https://docs.openebs.io/docs/next/snapshots.html).
+
 
 ## Clones using Kubectl
+Once you have created a snapshot, you can restore it to a new PVC. For this you must have a separate StorageClass implemented by snapshot-provisioner. Then you can create a PVC referencing to this StorageClass for dynamically provisioning new PersistentVolume.
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: For this = To do this

File: docs/tasks_snapclones.md [hunk 6]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+```
+kubectl get volumesnapshot
+```
+For more details on snapshots please click [here](https://docs.openebs.io/docs/next/snapshots.html).
+
 
 ## Clones using Kubectl
+Once you have created a snapshot, you can restore it to a new PVC. For this you must have a separate StorageClass implemented by snapshot-provisioner. Then you can create a PVC referencing to this StorageClass for dynamically provisioning new PersistentVolume. 
+
+You can create a separate StorageClass by using the the below contents in an yaml file. For example we have created 'restore-storageclass.yaml'
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: below = following contents
For example,

File: docs/tasks_snapclones.md [hunk 7]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: snapshot-promoter
+provisioner: volumesnapshot.external-storage.k8s.io/snapshot-promoter
+```
+
+Once the yaml has heen created, you have to deploy the yaml using the command below.
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: Once the yaml file is created, 
using the following command.

File: docs/tasks_snapclones.md [hunk 8]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+```
+kubectl apply -f restore-storageclass.yaml
+```
+
+Next you have to you have to create a PVC that will use the StorageClass to dynamically provision a PersistentVolume. You have to create a yaml file that will deploy a PVC using the below entries. Here for example we have created restore-pvc.yaml. The PVC that will be creaed here will be named as demo-snap-vol-claim
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: You must now create a PVC...
below entries = following entries.
For example, a restore-pvc.yaml is created.
creaed = created

File: docs/tasks_snapclones.md [hunk 9]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+    snapshot.alpha.kubernetes.io/snapshot: snapshot-demo
+spec:
+  storageClassName: snapshot-promoter
+  accessModes: ReadWriteOnce
+  resources:
+    requests:
+      storage: 5Gi
+```
+
+Then you can deploy the restore-pvc.yaml by using the command below.
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: You can then...

File: docs/tasks_snapclones.md [hunk 10]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+```   
+kubectl apply -f restore-pvc.yaml
+```
+
+Now you have to mount the demo-snap-vol-claim PVC into a similar application pod for which snapshot was taken. While deploying the pod, you have to mention the PVC name, volume naame, volume mount accordingly.
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: You must now mount...
naame = name

File: docs/tasks_snapclones.md [hunk 11]
```diff
@@ -6,8 +6,72 @@ sidebar_label: Snapshots and Clones
...
+
+For more details on clone please click [here](https://docs.openebs.io/docs/next/snapshots.html#cloning-and-restoring-a-snapshot).
```
Comment (mahebbar) [inline docs/tasks_snapclones.md]: For more details on cloning, click...

**Comment by ranjithwingrider** [comment]
Latest Doc change has been addressed with new steps.

**Pull request closed without merging**